bundle/edu.gemini.util.osgi/src/main/scala/edu/gemini/util/osgi/SecureServiceFactory.scala

```diff
@@ -1,52 +1,144 @@
+  final def getService(ps: Set[Principal]): A =
+    getService(ps.asJava)
 
-    def getSecureServices(clazz: String, filter: String, ps: Set[Principal]): List[(ServiceReference[_], Any)] = 
-      SecureServiceFactory.getSecureServices(bc, clazz, filter, ps)
+  def getService(ps: JSet[Principal]): A
 
-    def getSecureService(clazz: String, filter: String, ps: Set[Principal]): Option[(ServiceReference[_], Any)] = 
-      SecureServiceFactory.getSecureService(bc, clazz, filter, ps)
+}
+
+object SecureServiceFactory {
 
+  /**
+   * When we register a `SecureService[A]` with the container, the registration will have
+   * `ServiceProp` bound to the runtime classname of `A`, which is what we use for lookups.
+   */
+  val ServiceProp = "service"
+
+  /** Filter is a semigroup under conjunction. */
+  implicit val SemigroupFilter: Semigroup[Filter] =
+    Semigroup.instance((a, b) => createFilter(s"(&$a$b)"))
+
+  /** Construct an equality filter. */
+  def filter(key: String, value: String = "*"): Filter =
+    createFilter(s"($key=$value)")
+
+  /** Construct an filter binding `ServiceProp` to the name of the given class. */
+  def serviceFilter[A](clazz: Class[A]): Filter =
+    filter(ServiceProp, clazz.getName)
+
+  /**
+   * Look up a service factory for the given clazz with any additional filter conditions, create an
+   * instance with the given set of principals, evaluate the given function and return the result,
+   * releasing references as needed; or return `None` if the requested service is unavailable.
+   */
+  def withSecureService[A, B](
+    context: BundleContext,
+    clazz:   Class[A],
+    ps:      Set[Principal],
+    ofilter: Option[Filter]
+  )(f: A => B): Option[B] = {
+    val fcName = classOf[SecureServiceFactory[A]].getName
+    val filter = ofilter.foldLeft(serviceFilter(clazz))(_ |+| _).toString
+    val orefs  = Option(context.getServiceReferences(fcName,filter))
+    val oref   = orefs.flatMap(_.headOption)
+    oref.map { ref =>
+      val svc = context.getService(ref).asInstanceOf[SecureServiceFactory[A]]
+      try f(svc.getService(ps)) finally context.ungetService(ref)
+    }
   }
 
-}
 
-abstract class SecureServiceFactory[A <: AnyRef] extends ServiceFactory[A] {
+  def withSecureServiceByName[B](
+    context: BundleContext,
+    clazz:   String,
+    ps:      Set[Principal],
+    ofilter: Option[Filter]
+  )(f: Any => B): Option[B] =
+    withSecureService(context, Class.forName(clazz), ps, ofilter)(f)
+
+  /**
+   * Register the given `SecureServiceFactory` with the container, using any additional provided
+   * service properties, such that instances of `A` can be constructed via `withSecureService`
+   * above.
+   */
+  def registerSecureService[A](
+    context: BundleContext,
+    factory: SecureServiceFactory[A],
+    clazz:   Class[A],
+    props:   Map[String, String]
+  ): ServiceRegistration[SecureServiceFactory[A]] = {
+    val propsʹ = new JHashtable[String, Object]
```

Comment:
By accident there was a class that used the greek N in the name instead of the regular N. That is valid scala code and everything worked fine on spdb and dowloaded OT but when building the operational rpms for the OT (In the new build box) something, likely an encoding bug/feature produced an rpm that didn't work because of that.

Now this is a local variable so it should be less of a problem since the issue above exploded when the class was imported via the classloader